### ISSUE TITLE
Bootstrap patroni cluster from wal-g base backup

### ISSUE THREAD

**Issue by manish-k-sharma** [zalando/patroni#2319]
Hi,
I'm trying to restore a patroni cluster using the base backups created by another patroni cluster using wal-g to do restoration tests. I'm unable to restore this using only the base backup. I don't want to use the WAL files as they're huge.

Can someone suggest to me if it is even possible and if yes, what's the right configuration to do it?

My patroni configuration looks like below
```
method: bootstrap_with_walg
bootstrap_with_walg:
    command: /data/conf/wal-backup-fetch.sh
    keep_existing_recovery_conf: True
    recovery_conf:
        recovery_target_action: promote
        recovery_target_timeline: latest
        restore_command: /data/conf/wal-restore-command.sh "%f" "%p"
```

`wal-backup-fetch.sh` pulls the latest available wal-backup
```
wal-g backup-fetch /data/postgresql/current LATEST
```

`wal-restore-command` pull wal files
```
wal-g wal-fetch "$1" "$2"
```

The backups listed with wal-g are
```
name                          modified             wal_segment_backup_start
base_00000054000000FB000000D1 2022-05-23T05:36:03Z 00000054000000FB000000D1
```

I tried changing the `recovery_target_timeline` to different values but didn't help.
Also tried to remove the `recovery_conf` settings to bootstrap only from the base backup but the postgres server keeps failing. Also, tried changing the `restore_command` to `exit 0` but that didn't help too.

The most common error is below
```
t=2022-05-26 09:57:49.153 GMT u= db= pid=[594]: FATAL:  could not locate required checkpoint record
t=2022-05-26 09:57:49.153 GMT u= db= pid=[594]: HINT:  If you are restoring from a backup, touch "/data/postgresql/current/recovery.signal" and add required recovery options.
	If you are not restoring from a backup, try removing the file "/data/postgresql/current/backup_label".
```

Thanks in advance

**Comment by CyberDem0n** [comment]
> My patroni configuration looks like below

If it is correctly placed to a `bootstrap` section - that should work.

> Also tried to remove the recovery_conf settings to bootstrap only from the base backup but the postgres server keeps failing. Also, tried changing the restore_command to exit 0 but that didn't help too.

Well, that's expected. Postgres have to apply at some amount of WAL files on top of backup to reach conistency.

**Comment by manish-k-sharma** [comment]
@CyberDem0n thanks for confirming. It is correctly placed and patroni is able to trigger a recovery. What I'm not able to figure out is how can I bootstrap the cluster using base backup only and if it is even possible.

**Comment by manish-k-sharma** [comment]
Hi,

I understand that the WAL logs are needed to do PITR.
I managed to restore the cluster by deleting the backup_label and resetting the wals using the base backup only. I added below lines after wal-g backup-fetch.

rm /data/postgresql/current/backup_label
rm /data/postgresql/current/backup_label.old
mkdir /data/postgresql/current/pg_wal/archive_status
/usr/lib/postgresql/12/bin/pg_resetwal -f /data/postgresql/current/
Is there anything wrong with this approach? I'm happy to restore up to the point when base backup was initiated and lose the wal logs after that.

**Comment by CyberDem0n** [comment]
> Is there anything wrong with this approach?

Everything is wrong. Without WAL's backup is simply inconsistent.

**Comment by manish-k-sharma** [comment]
Noted. Thanks for confirmation.

**Comment by manish-k-sharma** [comment]
@CyberDem0n I reached the inconsistent state of the DB and appreciate your feedback.

I have a patroni-managed Postgres cluster. I've set up wal-g for creating backups in Google Cloud Storage. Everything works fine and I'm able to restore the cluster or bootstrap a new one using GCS bucket directly.

I need to ship these backups to tape and then prove that I can restore the backups from there. The problem is the volume of data. My base backups are being generated nightly and WAL archives are constantly filling up. The whole data has grown to TBs. Shipping this data to tape and restoring is posing challenges now.

I can copy a base backup from tape into cloud and restore but then the recovery keeps looking for WAL files. I'm unable to find a reasonable way of identifying the WAL files associated with a base backup so I can restore the database to a consistent state. And copying TBs of WAL files is certainly not feasible and ideal.

Can anyone suggest what is the general practice for restoring postgres from base backups and WAL files? My WAL directory is growing beyond bounds and is also a concern.
